astropy/visualization/tests/test_time.py

```diff
@@ -1,5 +1,7 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
+# ruff: noqa: E402
+
```

Comment:
Does it work if you simply put the `noqa: E402` comment on the line(s) triggering the issue?

Reply:
That could work too, but all the imports except `import pytest` would need their own `noqa` comments and I don't see what would be the benefit of doing that.

Reply:
Why not use the fact that `pytest.importorskip` returns the imported module if it is successful?

That is replace
```python
import matplotlib.dates
import matplotlib.pyplot as plt
```
with
```python
matplotlib = pytest.importorskip("matplotlib")
plt = matplotlib.pyplot
```
after all the imports (and letting ruff resort the imports)

I tested this locally, and this works as expected whether `matplotlib` is installed or not.

Reply:
`matplotlib` is needed here for two reasons.
1) To use `matplotlib.plt` in the tests: https://github.com/astropy/astropy/blob/1aa3e6a504090f401da647e3d7fc484888afd6e3/astropy/visualization/tests/test_time.py#L10
2) To set the value of the `MPL_EPOCH_1970` flag: https://github.com/astropy/astropy/blob/1aa3e6a504090f401da647e3d7fc484888afd6e3/astropy/visualization/tests/test_time.py#L16-L19
But the flag isn't needed anymore because `astropy` no longer supports `matplotlib` older than 3.3. I will open a separate pull request for removing it and when that is merged then it will be simpler to implement your suggestion for enforcing E402 also in this file.

Reply:
My suggestion does not require any of the changes you are referencing.

Reply:
It does not require them, but there will be less code churn if the other pull request is merged first.